### PR TITLE
Add additional check to check material library is loaded when trying to retrieve names

### PR DESCRIPTION
**Pull request by moraaar**
Cherry picked from development: https://github.com/o3de/o3de/pull/6050

### CODE REVIEW

**Review by amzn-sean — APPROVED** [review]
LGTM

File: Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp [hunk 1]
```diff
@@ -864,7 +864,8 @@ namespace PhysX
         {
             if (auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get())
             {
-                if (const auto* physicsConfiguration = physicsSystem->GetConfiguration())
+                if (const auto* physicsConfiguration = physicsSystem->GetConfiguration();
+                    physicsConfiguration && physicsConfiguration->m_materialLibraryAsset)
```
Comment (nemerle) [inline Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp:868]: Small nit: maybe a bit too much logic in this if? :)

Comment (moraaar) [reply Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp:868]: It's checking 2 pointers are valid using a variable limited to the scope of the `if`. It seems reasonable to me.

**Review by greerdv — APPROVED** [review]

File: Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp (hunk 1 above)
Comment (greerdv) [reply Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp:868]: I wasn't a huge fan of those initializer ifs when they were introduced, but I have come around to them. The scope limiting is nice.

Comment (nemerle) [reply Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp:868]: @greerdv I still find the 'old' style more readable :smile: 
```cpp
const auto* physicsConfiguration = physicsSystem->GetConfiguration();
if ( physicsConfiguration && physicsConfiguration->m_materialLibraryAsset)
{
```

**Review by nemerle — APPROVED** [review]
nits shouldn't have prevented me from marking as approved, sorry :smile:

File: Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp (hunk 1 above)
Comment (moraaar) [reply Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp:868]: Yeah, it took me some time to get used to it when introduced in C++17 too. But it's nice that the variable is available only where needed and not outside the `if`.

**Pull request merged**